cmd/kubelet/app/options/options.go

```diff
@@ -76,6 +76,11 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.TLSPrivateKeyFile, "tls-private-key-file", s.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
 	fs.StringVar(&s.CertDirectory, "cert-dir", s.CertDirectory, "The directory where the TLS certs are located (by default /var/run/kubernetes). "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
+	fs.StringVar(&s.BootstrapAuthToken, "bootstrap-auth-token", s.BootstrapAuthToken, ""+
+		"A string token that is used to get the issued certificate from API server. "+
+		"If --tls-cert-file, --tls-private-key-file are provided, then kubelet will use this token get a x509 cert from the API server"+
+		"that matches the generated private key."+
+		"The certificate and private key will be stored in --tls-cert-file and --tls-private-key-file.")
```

Comment:
If it's true that these are required (which it looks like they are right now), then there should be an earlier validation check before doing anything else that you've provided all three if you provided the token.

Is it not supported to just specify the token and have it request a new cert each time and not save it on disk though?